akka-remote/src/main/java/akka/remote/artery/compress/TopHeavyHittersScanning.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery.compress;
+
+public class TopHeavyHittersScanning {
+
+  private int _lowestHitterIdx = 0;
+  
+  protected final int[] hashes;
+  protected final Object[] items;
+  protected final long[] weights;
+
+  public TopHeavyHittersScanning(int max) {
+    hashes = new int[max];
+    weights = new long[max];
+    items = new Object[max];
+  }
+
+  protected boolean insertIfHeavy(Object item, int hashCode, long count) {
```

Comment:
why not expose any methods as public? I like to avoid inheritance. Or if you want to force inheritance for some reason, then I would prefer making these `protected final` and the fields it touches to be `private`.